test/e2e/control_plane_safe_eviction_test.go

```diff
@@ -0,0 +1,73 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	hyperv1 "github.com/openshift/hypershift/api/v1beta1"
+	cpoetcd "github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/etcd"
+	hcpmanifests "github.com/openshift/hypershift/hypershift-operator/controllers/manifests"
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+	corev1 "k8s.io/api/core/v1"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestControlPlaneSafeEviction(t *testing.T) {
+	t.Parallel()
+	g := NewWithT(t)
+
+	client, err := e2eutil.GetClient()
+	g.Expect(err).NotTo(HaveOccurred(), "failed to get k8s client")
+
+	ctx, cancel := context.WithCancel(testContext)
+	defer cancel()
+
+	clusterOpts := globalOpts.DefaultClusterOptions(t)
+
+	hostedCluster := e2eutil.CreateCluster(t, ctx, client, &clusterOpts, globalOpts.Platform, globalOpts.ArtifactDir, globalOpts.ServiceAccountSigningKey)
+
+	// Get the newly created NodePool
+	nodepools := &hyperv1.NodePoolList{}
+	if err := client.List(ctx, nodepools, crclient.InNamespace(hostedCluster.Namespace)); err != nil {
+		t.Fatalf("failed to list nodepools in namespace %s: %v", hostedCluster.Namespace, err)
+	}
+	if len(nodepools.Items) != 1 {
+		t.Fatalf("expected exactly one nodepool, got %d", len(nodepools.Items))
+	}
+	nodepool := &nodepools.Items[0]
+
+	guestClient := e2eutil.WaitForGuestClient(t, ctx, client, hostedCluster)
+	numNodes := clusterOpts.NodePoolReplicas
+	nodes := e2eutil.WaitForNReadyNodes(t, ctx, guestClient, numNodes, hostedCluster.Spec.Platform.Type)
+
+	// Wait for the rollout to be reported complete
+	t.Logf("Waiting for cluster rollout. Image: %s", globalOpts.LatestReleaseImage)
+	e2eutil.WaitForImageRollout(t, ctx, client, hostedCluster, globalOpts.LatestReleaseImage)
+
+	// Check if PodSafeToEvict annotation exists in hcp pods
+	hcpNs := hcpmanifests.HostedControlPlaneNamespace(hcluster.Namespace, hcluster.Name).Name
+	hcpPods := &corev1.PodList{}
+	if err := r.List(ctx, hcpPods, &client.ListOptions{
+		Namespace: hcpNs,
+	}); err != nil {
+		return fmt.Errorf("cannot list hostedControlPlane pods: %w", err)
+	}
+
+	// Check if the pods with PVC or emptyDir mounts are annotated with safe-to-evict annotation from CA operator
```

Comment:
why PVC?

Reply:
I meant volumes, I've fixed in my working on version.

Reply:
not sure I follow, the ones we want to annotate are just those with an emptyDir volume https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#what-types-of-pods-can-prevent-ca-from-removing-a-node
The if below seems wrong to me.

Reply:
> Any reason why this PR is changing etcd? how about all the other components mentioned in https://issues.redhat.com/browse/OCPBUGS-11882 that we manage?

nope, just a misunderstood.